calyx-opt/src/passes/simplify_static_guards.rs

```diff
@@ -0,0 +1,141 @@
+use crate::traversal::{Action, Named, VisResult, Visitor};
+use calyx_ir as ir;
+use calyx_ir::Interval;
+
+#[derive(Default)]
+/// Simplifies Static Guards
+/// In particular if g = g1 & g2 & ...gn, then it takes all of the g_i's that
+/// are "static timing intervals", e.g., %[2:3], and combines them into one
+/// timing interval.
+/// For example: (port.out | !port1.out) & (port2.out == port3.out) & %[2:8] & %[5:10] ?
+/// becomes (port.out | !port1.out) & (port2.out == port3.out) & %[5:8] ?
+/// by "combining" %[2:8] & %[5:10]
+pub struct SimplifyStaticGuards;
```

Comment:
What happens if I have something like:
```
(%[2:8] & %[5:10]) | (%[3:5])
```

Are disjunctions supported? In this case we can merge them into one guard but when we have other guards floating around, that might be tricky